### PR TITLE
Adding codebuild_initiator to a pipeline name

### PR DESCRIPTION
**Pull request by SergTogul**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
There's a number of different MAINLINE pipelines those overriding each other test executions results. Add codebuild_initiator to a pipeline name for distinguishing different build pipelines those may be triggered with the same commit id.

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/testrunner.py [hunk 1]
```diff
@@ -259,6 +260,15 @@ def main():
     benchmark_mode = "benchmark" in test_type or is_benchmark_dev_context()
     specific_test_type = re.sub("benchmark-", "", test_type) if "benchmark" in test_type else test_type
     build_context = get_build_context()
+    # PR test executions can be distinguished by commit id. 
+    # But other pipelines triggered with the same commit id. 
+    # Adding codebuild_initiator to a pipeline name for distinguishing different build pipelines 
+    if not "PR" == build_context:
```
Comment (arjkesh) [inline test/testrunner.py:266]: There should be a function called `is_pr_context()` - however, are there specific contexts where this condition should apply? Should this also run in canary, for instance, or just MAINLINE and NIGHTLY?

File: test/testrunner.py [hunk 2]
```diff
@@ -259,6 +260,15 @@ def main():
...
+        codebuild_initiator = get_codebuild_initiator()
+        if codebuild_initiator:
+            build_context += f"_{codebuild_initiator}"
```
Comment (arjkesh) [inline test/testrunner.py:269]: I don't think we should be changing `build_context` like this - as it's also used for eks cluster creation, plus adding codebuild initiator isn't really a part of the build context as we've defined it in different parts of the code. 

Can we add a new param to pytest_cache_params, or define a different variable for this?

File: test/testrunner.py (hunk 1 above)
Comment (arjkesh) [reply test/testrunner.py:266]: Additionally, I'm not sure I totally understand the reason for not considering the PR context. If we add the ability to run multiple types of jobs in parallel (i.e. graviton alongside regular cpu/gpu), then we'll run into the same problem again, right?

Comment (SergTogul) [reply test/testrunner.py:266]: PR jobs are triggered with different commit IDs. There won't be 2 PR jobs with the same commit ID. So we don't have to do it for PR pipelines

Comment (arjkesh) [reply test/testrunner.py]: We should keep it consistent. If we add the ability to run graviton and cpu/gpu from the same commit, for example, we'll have to retroactively track this down and remove this condition.

**Pull request closed without merging**